com.unity.formats.fbx/Editor/ExportModelSettings.cs

```diff
@@ -164,8 +188,25 @@ internal abstract class ExportOptionsSettingsSerializeBase : IExportOptions
         public abstract ExportSettings.LODExportType LODExportType { get; }
         public abstract ExportSettings.ObjectPosition ObjectPosition { get; }
         public abstract bool ExportUnrendered { get; }
-        public virtual bool PreserveImportSettings { get;  }
+        public virtual bool PreserveImportSettings { get { return false; } }
         public abstract bool AllowSceneModification { get; }
+
+        public override bool Equals(object e)
+        {
+            var expOptions = e as ExportOptionsSettingsSerializeBase;
+            if (expOptions == null)
+            {
+                return false;
+            }
+            return animatedSkinnedMesh == expOptions.animatedSkinnedMesh &&
+                mayaCompatibleNaming == expOptions.mayaCompatibleNaming &&
+                exportFormat == expOptions.exportFormat;
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```

Comment:
This is invalid: it can return different hash codes on two items that test equal.

Given you're basically got three bools, the hash code can be the bitmask of the three bools.